src/main/java/org/broadinstitute/hellbender/engine/ContextShard.java

```diff
@@ -0,0 +1,91 @@
+package org.broadinstitute.hellbender.engine;
+
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.collections.IntervalsSkipListOneContig;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+
+/**
+ * Immutable storage class.
+ * This class enforces no consistency guarantees. The intent of course
+ * is for the reads and variants to be in the interval, but that's up the callers.
```

Comment:
Is this really the intent? Eg., when you add the variants in `fillVariants()` via `ret.add(new ContextShard(s).withVariants(intervals.getOverlapping(interval)));`, many variants that don't actually overlap the shard's interval are added due to margin expansion. Is there a need to track the actual interval that the inserted records overlap?